### ISSUE TITLE
Money-math on browser via browserify

### ISSUE THREAD

**Issue by mstn** [ikr/money-math#3]
Hi, thanks for your work. I like this library because its api is clean. I am trying to use it on browser using browserify and I noticed it includes some binary blobs for bignum. As far I know, there is no way to make it work on browser. Have you considered to think about a browser-friendly version without binaries?

**Comment by ikr** [comment]
Hi Marco! I'm glad you like it. Correct; I already answered that before via e-mail, actually :) Will be better to save it here in an issue:

```
Hi Steven

I'm afraid there's no easy way. money-math relies on a native Node extension which
uses a C-library for arbitrary precision math. Theoretically, one could compile that
library to JS with Emscripten, but that's way too hardcore. Instead, check jsbn.js out:

http://www-cs-students.stanford.edu/~tjw/jsbn/

-- it's a BigInteger implementation in pure JS. I should probably consider porting
money-math to jsbn back-end... Pull requests are welcome ;)

Regards
Ivan

On 06.12.2014, at 02:35, Steven Walters <steve...@gmail.com> wrote:

> Hello,
>
> Is there anyway to use money-math with stock JavaScript and jQuery?
> Or is node.js required? I'm just getting my head around JavaScript and
> jQuery and I'm not sure I am ready to tackle node.js yet.
>
> Thank you,
> --StevenKW
```

**Comment by mstn** [comment]
Hi Ivan, thanks for your answer. When I have a bit of time (I do not think soon), I'll give a look into it. Perhaps the best thing to do is to ditch bignum in favor of jsbn. Then it should be easy to make this library work also for the browser using browserify.

**Comment by ikr** [comment]
Gotta measure the performance. For a serious server-side number crunching, the native bignum may appear to be way faster. Thus, having the both back-ends may be necessary.

**Comment by ikr** [comment]
The just published v2.0.0 is browser-compatible. I've switched from bignum to jsbn back-end. It appeared that [performance is fine](https://github.com/asdf404/bignum-bench)

**Comment by mstn** [comment]
Thanks!
